graylog2-web-interface/src/components/extractors/extractors_configuration/SplitAndIndexExtractorConfiguration.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useCallback, useEffect, useState } from 'react';
+
+import { Col, Row, Button } from 'components/graylog';
+import { Icon } from 'components/common';
+import { Input } from 'components/bootstrap';
+import UserNotification from 'util/UserNotification';
+import ExtractorUtils from 'util/ExtractorUtils';
+import FormUtils from 'util/FormsUtils';
+import ToolsStore from 'stores/tools/ToolsStore';
+
+const DEFAULT_CONFIGURATION = { index: 1 };
+
+const _getEffectiveConfiguration = (configuration) => ExtractorUtils.getEffectiveConfiguration(DEFAULT_CONFIGURATION, configuration);
+
+type Configuration = { [key: string]: string };
+type Props = {
+  configuration: Configuration,
+  exampleMessage: string,
+  onChange: (newConfig: Configuration) => void,
+  onExtractorPreviewLoad: (preview: React.ReactNode | string) => void,
+}
+
+const SplitAndIndexExtractorConfiguration = ({ configuration: initialConfiguration, exampleMessage, onChange, onExtractorPreviewLoad }: Props) => {
+  const [configuration, setConfiguration] = useState(_getEffectiveConfiguration(initialConfiguration));
+  useEffect(() => { setConfiguration(_getEffectiveConfiguration(initialConfiguration)); }, [initialConfiguration]);
+
+  const [trying, setTrying] = useState(false);
+
+  const _onChange = (key: string) => {
+    return (event) => {
+      onExtractorPreviewLoad(undefined);
+      const newConfig = configuration;
+
+      newConfig[key] = FormUtils.getValueFromInput(event.target);
+      onChange(newConfig);
+    };
+  };
+
+  const _onTryClick = useCallback(() => {
+    setTrying(true);
+
+    const promise = ToolsStore.testSplitAndIndex(configuration.split_by, configuration.index, exampleMessage);
+
+    promise.then((result) => {
+      if (!result.successful) {
+        UserNotification.warning('We were not able to run the split and index extraction. Please check your parameters.');
+
+        return;
+      }
+
+      const preview = (result.cut ? <samp>{result.cut}</samp> : '');
+
+      onExtractorPreviewLoad(preview);
+    });
+
+    promise.finally(() => setTrying(false));
+  }, []);
```

Comment:
Currently we are always calling `ToolsStore.testSplitAndIndex` with the initial form values, since there are no dependencies for `useCallback`.

Reply:
Good catch!